aws/resource_aws_lb.go

```diff
@@ -638,6 +638,17 @@ func flattenSubnetMappingsFromAvailabilityZones(availabilityZones []*elbv2.Avail
 			l = append(l, m)
 		}
 	}
+
+	if v, ok := d.GetOk("subnet_mapping"); ok {
```

Comment:
Using Terraform state information to determine setting other Terraform state information is unfortunately an anti-pattern. While it may work in some simple scenarios, there are generally problems associated with this technique and it does not work with `terraform import`.

In this case, the logic above this addition here just needed to always set the `subnet_id` in the outer `for` loop instead of depending on the inner `for` loop. You can see an example of this change here: https://github.com/terraform-providers/terraform-provider-aws/pull/8282/files#diff-5937d84fb951c53ec8952ec34df1feacL630